File: python/704_binary_search.py
class Solution(object):
    def search(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        low = 0
        high = len(nums)

        while low < high:
            cur = (low + high) // 2
            if nums[cur] == target:
                return cur
            elif nums[cur] < target:
                low = cur + 1
            else:
                high = cur
        
        return -1

File: c++/704_binary_search.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int l = 0, r = nums.size() - 1;

        while (l <= r) {
            int ind = (r + l) / 2;
            if (nums[ind] == target) {
                return ind;
            } else if (nums[ind] < target) {
                l = ind + 1;
            } else {
                r = ind - 1;
            }
        }

        return -1;
    }
};
